Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterLauncher.cs

```diff
@@ -0,0 +1,213 @@
+
+            DebugLaunchInfo debugLaunchInfo = new DebugLaunchInfo() {
+                Cwd = adapterLaunchInfoJson.Value<string>("cwd"),
+                Console = "externalTerminal",
+            };
+
+            SetInterpreterOptions(debugLaunchInfo, adapterLaunchInfoJson);
+            SetProgramInfo(debugLaunchInfo, adapterLaunchInfoJson);
+            SetEnvVariables(debugLaunchInfo, adapterLaunchInfoJson);
+            SetLaunchDebugOptions(debugLaunchInfo, adapterLaunchInfoJson);
+
+            return debugLaunchInfo;
+
+        }
+        private static void SetInterpreterOptions(DebugLaunchInfo debugLaunchInfo, JObject adapterLaunchInfoJson) {
+            debugLaunchInfo.Python = new List<string>() {
+                adapterLaunchInfoJson.Value<string>("exe").Replace("\"", "")
+            };
+
+            string interpreterArgs = adapterLaunchInfoJson.Value<string>("interpreterArgs");
+            debugLaunchInfo.Python.AddRange(GetParsedCommandLineArguments(interpreterArgs));
+        }
+        private static void SetProgramInfo(DebugLaunchInfo debugLaunchInfo, JObject adapterLaunchInfoJson) {
+            debugLaunchInfo.Program = new List<string> {
+                adapterLaunchInfoJson.Value<string>("scriptName")
+            };
+
+            string scriptArgs = adapterLaunchInfoJson.Value<string>("scriptArgs");
+            debugLaunchInfo.Program.AddRange(GetParsedCommandLineArguments(scriptArgs));
+        }
+        private static void SetEnvVariables(DebugLaunchInfo debugLaunchInfo, JObject adapterLaunchInfoJson) {
+            var env = new Dictionary<string, string>();
+            foreach (var envVariable in adapterLaunchInfoJson.Value<JArray>("env")) {
+                env[envVariable.Value<string>("name")] = envVariable.Value<string>("value");
+            }
+
+            debugLaunchInfo.Env = env.Count == 0 ? null : env;
+        }
+        private static void SetLaunchDebugOptions(DebugLaunchInfo debugLaunchInfo, JObject adapterLaunchInfoJson) {
+            string[] options = SplitDebugOptions(adapterLaunchInfoJson.Value<string>("options"));
+
+            string djangoOption = options.FirstOrDefault(x => x.StartsWith("DJANGO_DEBUG"));
+            if (djangoOption != null) {
+                string[] parsedOption = djangoOption.Split('=');
+                if (parsedOption.Length == 2) {
+                    debugLaunchInfo.DebugDjango = parsedOption[1].ToLower().Trim().Equals("true");
+                }
+            }
+
+            string webPageUrlOption = options.FirstOrDefault(x => x.StartsWith("WEB_BROWSER_URL"));
+            if (webPageUrlOption != null) {
+                string[] parsedOption = webPageUrlOption.Split('=');
+                if (parsedOption.Length == 2) {
+                    debugLaunchInfo.WebPageUrl = WebUtility.UrlDecode(parsedOption[1].ToLower().Trim());
```

Comment:
why lower the url? the user will notice this in the browser. is it new or was that done previously?

Reply:
Was not done previously--has been reverted.